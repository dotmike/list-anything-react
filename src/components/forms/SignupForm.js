import React from "react";
import PropTypes from "prop-types";
import { Form, Button, Segment } from "semantic-ui-react";
import isEmail from "validator/lib/isEmail";

import SimpleForm from "./SimpleForm";
import InlineError from "../messages/InlineError";
import ErrorMessage from "../messages/ErrorMessage";

class SignupForm extends React.Component {
  getData = getInputByName => ({
    email: getInputByName("email").value,
    password: getInputByName("password").value
  });

  validate = data => {
    const errors = {};
    if (!isEmail(data.email)) errors.email = "Invalid email";
    if (!data.password) errors.password = "Can't be blank";
    return errors;
  };

  renderFormData = ({ errors }) => (
    <React.Fragment>
      {errors.global && <ErrorMessage text={errors.global} />}
      <Segment stacked>
        <Form.Field error={!!errors.email}>
          <Form.Input
            fluid
            icon="user"
            iconPosition="left"
            type="email"
            placeholder="example@example.com"
            name="email"
          />
          {errors.email && <InlineError text={errors.email} />}
        </Form.Field>

        <Form.Field error={!!errors.password}>
          <Form.Input
            fluid
            icon="lock"
            iconPosition="left"
            placeholder="Password"
            type="password"
            name="password"
          />
          {errors.password && <InlineError text={errors.password} />}
        </Form.Field>
        <Button color="teal" fluid size="large">
          Sign up
        </Button>
      </Segment>
    </React.Fragment>
  );

  render() {
    return (
      <SimpleForm
        id="signup-form"
        validate={this.validate}
        render={this.renderFormData}
        getData={this.getData}
        submit={this.props.submit}
      />
    );
  }
}

SignupForm.propTypes = {
  // ownProps
  submit: PropTypes.func.isRequired
};

export default SignupForm;
