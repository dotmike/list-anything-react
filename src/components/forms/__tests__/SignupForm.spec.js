import React from "react";
import { shallow } from "enzyme";

import SignupForm from "../SignupForm";

const defaultProps = {
  submit: () => {}
};

describe("SignupForm", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<SignupForm {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
