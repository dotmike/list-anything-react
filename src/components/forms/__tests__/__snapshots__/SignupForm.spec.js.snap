// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupForm renders correctly 1`] = `
<SimpleForm
  getData={[Function]}
  id="signup-form"
  render={[Function]}
  submit={[Function]}
  validate={[Function]}
/>
`;
